src/taskpane/components/ReferenceList.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  ITheme,
+  getTheme,
+  List,
+  mergeStyleSets,
+  getFocusStyle,
+  FocusZone,
+  FocusZoneDirection,
+  Checkbox,
+} from "@fluentui/react";
+import React from "react";
+
+const theme: ITheme = getTheme();
+const { palette, semanticColors, fonts } = theme;
+
+interface ReferenceListProps {
+  list: Array<{}>;
+  onCheckBoxChange: any;
+}
+const classNames = mergeStyleSets({
+  container: {
+    overflow: "auto",
+    height: "80vh",
+    padding: "0.25rem 0.25rem 0px",
+    boxSizing: "border-box",
+  },
+  itemCell: [
+    getFocusStyle(theme, { inset: -1 }),
+    {
+      backgroundColor: theme.palette.neutralLighterAlt,
+      minHeight: 54,
+      padding: "0.25rem",
+      margin: 2,
+      boxSizing: "border-box",
+      borderBottom: `1px solid ${semanticColors.bodyDivider}`,
+      display: "flex",
+      selectors: {
+        "&:hover": { background: palette.themeLighterAlt },
+      },
+    },
+  ],
+  itemContent: {
+    marginLeft: 10,
+    boxSizing: "border-box",
+    overflow: "auto",
+    flexGrow: 1,
+  },
+  itemTitle: [
+    fonts.mediumPlus,
+    {
+      whiteSpace: "nowrap",
+      position: "relative",
+      maxHeight: "5.4em",
+      lineHeight: "1.8em",
+      overflow: "hidden",
+      textOverflow: "ellipsis",
+    },
+  ],
+  itemAuthor: {
+    fontSize: fonts.small.fontSize,
+    color: palette.neutralTertiary,
+    marginBottom: 10,
+  },
+  itemYear: {
+    fontSize: fonts.smallPlus,
+    color: palette.neutralTertiary,
+  },
+  itemType: {
+    fontSize: fonts.smallPlus,
+    color: palette.neutralTertiary,
+  },
+  checkbox: {
+    marginTop: 6,
+    marginLeft: 4,
+  },
+});
+
+function ReferenceList(props: ReferenceListProps) {
+  const onRenderCell = (item): JSX.Element => {
```

Comment:
Again, this might be a result of my unfamiliarity with react, but doesn't it make sense to extract this code to a new method `ReferenceItem` that takes care of the display of one item?

Reply:
Fluent UI  :/
This function can only accept a set of parameters defined in the List's ` onRenderCell`. But we also have to pass the event handler function for the checkbox. That's why I kept it inside the ReferenceList component so that eventhandlerfunction can be passed directly from props.